pyo3-macros-backend/src/params.rs

```diff
@@ -9,6 +9,56 @@ use quote::{quote, quote_spanned};
 use syn::spanned::Spanned;
 use syn::Result;
 
+pub struct Holders {
+    holders: Vec<syn::Ident>,
+    gil_refs_checkers: Vec<syn::Ident>,
+}
+
+impl Holders {
+    pub fn new() -> Self {
+        Holders {
+            holders: Vec::new(),
+            gil_refs_checkers: Vec::new(),
+        }
+    }
+
+    pub fn push_holder(&mut self, span: Span) -> syn::Ident {
+        let holder = syn::Ident::new(&format!("holder_{}", self.holders.len()), span);
+        self.holders.push(holder.clone());
+        holder
+    }
+
+    pub fn push_gil_refs_checker(&mut self, span: Span) -> syn::Ident {
+        let gil_refs_checker = syn::Ident::new(
+            &format!("gil_refs_checker_{}", self.gil_refs_checkers.len()),
+            span,
+        );
+        self.gil_refs_checkers.push(gil_refs_checker.clone());
+        gil_refs_checker
+    }
+
+    pub fn init_holders(&self, ctx: &Ctx) -> TokenStream {
+        let Ctx { pyo3_path } = ctx;
+        let holders = &self.holders;
+        let gil_refs_checkers = &self.gil_refs_checkers;
+        quote! {
+            #[allow(clippy::let_unit_value)]
+            #(let mut #holders = #pyo3_path::impl_::extract_argument::FunctionArgumentHolder::INIT;)*
+            #(let #gil_refs_checkers = #pyo3_path::impl_::deprecations::GilRefs::new();)*
+        }
+    }
+
+    pub fn check_gil_refs(&self) -> TokenStream {
+        let gil_refs_checkers = self
+            .gil_refs_checkers
+            .iter()
+            .map(|e| quote_spanned! { e.span() => #e.function_arg(); });
+        quote! {
+            #(#gil_refs_checkers)*
+        }
```

Comment:
```suggestion
       self.gil_refs_checkers
            .iter()
            .map(|e| quote_spanned! { e.span() => #e.function_arg(); })
            .collect()
```

I think you can also just `.collect()` into a `TokenStream`.